### PR TITLE
Download and process plates

### PR DESCRIPTION
**Pull request by MattsonCam**
In this pr, the plate data is downloaded from aws and then processed with pycytominer using reference plate data. The code wasn't rerun to avoid the long process of recreating the large datasets. Please let me know if you have any questions.

@jenna-tomkinson contributed to this pr.

### CODE REVIEW

File: 0.download_data/download_from_aws.sh
Comment (gwaybio) [inline 0.download_data/download_from_aws.sh]:
```suggestion
    # Download the s3 file to your local machine in a directory defined by command line argument
```

File: 0.download_data/jump_dataset.csv
Comment (gwaybio) [inline 0.download_data/jump_dataset.csv]: what is the provenance of this file? (How was this file generated?)

You can specify this provenance in a README :)

Comment (gwaybio) [reply 0.download_data/jump_dataset.csv]: Also, are these all `COMPOUND` plates? Are there other `CRISPR` and `ORF`?

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (gwaybio) [inline 1.process_data/nbconverted/process_and_normalize_jump_data.py]: Functions should be defined at the beginning of the notebook

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (gwaybio) [inline 1.process_data/nbconverted/process_and_normalize_jump_data.py]:
```suggestion
    df.rename(columns=lambda x: f"Metadata_{x}", inplace=True)
```

Updating to f-strings

Comment (gwaybio) [reply 1.process_data/nbconverted/process_and_normalize_jump_data.py]: Also, is there a way to avoid setting `Metadata_Metadata_columnname`? Does already having a `Metadata_` prefix ever occur?

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (gwaybio) [inline 1.process_data/nbconverted/process_and_normalize_jump_data.py]:
```suggestion
def add_metadata_prefix_to_column_names(df):
```

I recommend using very explicit function names :)

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (gwaybio) [inline 1.process_data/nbconverted/process_and_normalize_jump_data.py:163]: Love this!

File: environment.yml
Comment (gwaybio) [inline environment.yml]: this is an incredibly strict conda environment that is not likely to work on any computer other than your own.

I recommend using a simplified conda environment, which specifies only essential packages and versions.

If you would like to go the very specific environment route, then you should use docker.

File: environment.yml
Comment (gwaybio) [inline environment.yml]: what is `prefix`? Note, this conda environment should work for other computers and users

**Review by gwaybio — CHANGES_REQUESTED** [review]
A couple of in-line comments.

It's worth adding a README to the `0.download_data` and `1.process_data` modules to explicitly describe your approach

File: 0.download_data/download_from_aws.sh
Comment (MattsonCam) [reply 0.download_data/download_from_aws.sh]: Good catch, changed!

File: 0.download_data/jump_dataset.csv
Comment (MattsonCam) [reply 0.download_data/jump_dataset.csv]: These include all of the plates. I could also include the code for generating this file, but it essentially just creates the s3 links with the plate names and the aws file path.

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (MattsonCam) [reply 1.process_data/nbconverted/process_and_normalize_jump_data.py]: Sounds good, these are at the beginning after the imports and what not

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (MattsonCam) [reply 1.process_data/nbconverted/process_and_normalize_jump_data.py]: Yes, you could check if the column name already contains the prefix, but this doesn't happen.

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (MattsonCam) [reply 1.process_data/nbconverted/process_and_normalize_jump_data.py]: Changed!

File: environment.yml
Comment (MattsonCam) [reply environment.yml]: That is good to know, I always just exported my conda environment to a yaml file. After talking with Mike and Jenna, I think I will manually specify the packages I install in the conda environment file. This isn't a perfect solution, so I'm open to suggestions.

File: environment.yml
Comment (MattsonCam) [reply environment.yml]: I think the `prefix` is a consequence of exporting the environment. Please see the comment above.

**Comment by MattsonCam** [comment]
Hey @gwaybio, this analysis should be updated as well. Please let me know if you have any other feedback.

File: 0.download_data/jump_dataset.csv
Comment (gwaybio) [reply 0.download_data/jump_dataset.csv]: Adding code that generates this file is important for reproducibility :)

It is good practice to also include details on how you executed the file (e.g., explicit run-time instructions)

File: environment.yml
Comment (gwaybio) [reply environment.yml]: manually specifying the packages is the way to go. It's a trade-off between rigid environments that don't work on many computers and loose environments that fail to reproduce the analysis. I recommend moving a bit towards the loose environment.

File: 0.download_data/download_jump_data.py
Comment (gwaybio) [inline 0.download_data/download_jump_data.py]: When you `nbconvert` these files, please put them in a subfolder so as not to confuse someone looking at your repository.

I typically put them in a folder called `nbconverted`.

File: 0.download_data/jump_dataset.csv
Comment (gwaybio) [inline 0.download_data/jump_dataset.csv]: I have three additional comments on this file.

1. The name is not very specific. Is there something you can think of that would be more accurate? Maybe `jump_dataset_location_manifest.csv`?
2. When you're creating these files, it is super important to have column names. This will tell everyone (including future you!) what each of these columns represent
3. Please place this file inside a folder (maybe `data`?) this will help to organize the repo and make it more readable

File: 0.download_data/download_jump_data.ipynb
Comment (gwaybio) [inline 0.download_data/download_jump_data.ipynb]: If you add the file you generated `jump_dataset.csv` with, I recommend also changing the name of this file to include a sequential prefix.

E.g.
0.generate_jump_dataset_manifest.ipynb
1.download_jump_data.ipynb

(also, shouldn't `1.download_jump_data.ipynb` actually be named `1.download_jump_single_cell_profiles.ipynb`? Someone else might think that you're downloading the images and you're not.

File: 0.download_data/jump_dataset.csv
Comment (gwaybio) [inline 0.download_data/jump_dataset.csv]: I still see only `COMPOUND` platemap types. Shouldn't there be others (e.g. CRISPR)?

File: README.md
Comment (gwaybio) [inline README.md]:
```suggestion
Single cell analysis of the JUMP Cell Painting consortium data.
Please ensure adequate storage as the sqlite plate data is approximately 1.1 TB large.
```

We use one-sentence-per-line in our markdown files (nit-picky, I know!) :)

File: README.md
Comment (gwaybio) [inline README.md]:
```suggestion
## 0.download_data

This folder contains the code and metadata file to download the plate data from aws.
The metadata contains the following information:
```

Minor formatting suggestions

File: README.md
Comment (gwaybio) [inline README.md]:
```suggestion
## 1.process_data

In this step, the data was merged into single cell data using pycytominer, and the reference plate data located in the `reference_plate_data` folder.
Then the plate data was normalized by pycytominer and converted to parquet files.
```

markdown formatting

File: environment.yml
Comment (gwaybio) [inline environment.yml]: Many of these packages are available on conda-forge (pathlib, numpy, pandas, matplotlib, scipy, scikit-learn). The only one to download this way is pycytominer.

See the change log and make sure you install the primary python packages through conda-forge.

It is good practice to limit pip usage in a conda environment: https://www.anaconda.com/blog/using-pip-in-a-conda-environment

File: README.md
Comment (gwaybio) [reply README.md]: It is probably also worth adding the provenance of the `reference_plate_data` in this README under `1.process_data`. Where did you get these files from?

**Review by gwaybio — CHANGES_REQUESTED** [review]
More comments coming your way!

Please let me know if you have any questions. Nearly ready to merge

File: environment.yml
Comment (MattsonCam) [reply environment.yml]: I agree, these changes should now be reflected.

File: 0.download_data/jump_dataset.csv
Comment (MattsonCam) [reply 0.download_data/jump_dataset.csv]: Sounds good, just added this!

File: 0.download_data/download_jump_data.py
Comment (MattsonCam) [reply 0.download_data/download_jump_data.py]: That works, I added this

File: 0.download_data/jump_dataset.csv
Comment (MattsonCam) [reply 0.download_data/jump_dataset.csv]: 1. That sounds like a good name, I updated the file.
2. That makes sense, this is also updated
3. Changed!

File: 0.download_data/download_jump_data.ipynb
Comment (MattsonCam) [reply 0.download_data/download_jump_data.ipynb]: Good point, renamed!

File: 0.download_data/jump_dataset.csv
Comment (MattsonCam) [reply 0.download_data/jump_dataset.csv]: I removed this column, I created it arbitrarily at first and realized later it is not useful.

File: README.md
Comment (MattsonCam) [reply README.md]: Changed!

File: README.md
Comment (MattsonCam) [reply README.md]: Changed!

File: README.md
Comment (MattsonCam) [reply README.md]: Sounds good, this has been updated

File: environment.yml
Comment (MattsonCam) [reply environment.yml]: That's good to know, the conda environment file should be updated now.

**Comment by MattsonCam** [comment]
Hey @gwaybio, just finished reviewing some of your useful comments. This was a larger pr, so feel free to let me know if you have any others

**Comment by MattsonCam** [comment]
I should almost be done predicting the single cell probabilities, and then I'll create another pr. Otherwise, I think I've addressed the other comments for this pr. If you have any additional comments before approval @gwaybio, feel free to add them.

File: 0.download_data/.jukit/.jukit_info.json
Comment (gwaybio) [inline 0.download_data/.jukit/.jukit_info.json]: I think this file (and other `.jukit` files) was added by mistake? Consider adding `.jukit` to gitignore.

File: 0.download_data/0.generate_jump_dataset_manifest.py
Comment (gwaybio) [inline 0.download_data/0.generate_jump_dataset_manifest.py]: what is the provenance of these object names? Is there a way to make this more programmatic? I bet you can find these in a metadata file somewhere in the JUMP repo?

Making this programmatic will help us in the future, when we want to run this approach on more JUMP data (remember, this is just the pilot)

File: 0.download_data/1.download_jump_single_cell_profiles.py
Comment (gwaybio) [inline 0.download_data/1.download_jump_single_cell_profiles.py]: I think you have recently moved this to `nbconverted` (which is where it should go) but maybe haven't deleted this file?

File: 0.download_data/nbconverted/download_jump_data.py
Comment (gwaybio) [inline 0.download_data/nbconverted/download_jump_data.py]: I think you've since added a prefix to this notebook, but haven't yet deleted this file?

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (gwaybio) [inline 1.process_data/nbconverted/process_and_normalize_jump_data.py]: It looks like this notebook isn't executed (or at least it wasn't nbconverted after execution)

**Review by gwaybio — APPROVED** [review]
Please merge after you address the comments. They mainly have to do with some minor gardening:

1. Adding to gitignore
2. Making sure to remove redundant files (removing some files in `nbconverted` and from `0.download_data` that you have since renamed)

No need for me to take another look once you are able to do this!

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (MattsonCam) [reply 1.process_data/nbconverted/process_and_normalize_jump_data.py]: I'll rerun this code to show the outputs, and then repush

File: 0.download_data/nbconverted/download_jump_data.py
Comment (MattsonCam) [reply 0.download_data/nbconverted/download_jump_data.py]: Ya, I changed the notebook filename, so this isn't needed anymore, deleted!

File: 0.download_data/1.download_jump_single_cell_profiles.py
Comment (MattsonCam) [reply 0.download_data/1.download_jump_single_cell_profiles.py]: This was fixed with the gitignore, this is a python jukit file, similar to a jupyter notebook but for python

File: 0.download_data/.jukit/.jukit_info.json
Comment (MattsonCam) [reply 0.download_data/.jukit/.jukit_info.json]: You're right, I thought I copied my previous .gitignore, thanks for catching this.

File: 0.download_data/0.generate_jump_dataset_manifest.py
Comment (MattsonCam) [reply 0.download_data/0.generate_jump_dataset_manifest.py]: Just changed this, should now be updated in the repo

File: 1.process_data/nbconverted/process_and_normalize_jump_data.py
Comment (MattsonCam) [reply 1.process_data/nbconverted/process_and_normalize_jump_data.py]: Just pushed the reran code

**Comment by MattsonCam** [comment]
Thanks @gwaybio and @jenna-tomkinson, I'm ready to merge. Please feel free to merge whenever ready @gwaybio.

**Pull request merged**
